### DISCUSSION TITLE
Pocketbase React / Native / Expo SDK

### DISCUSSION THREAD

**Discussion by tobicrain** [pocketbase/pocketbase#706] [Show and tell]
I made some kind of wrapper for Pocketbase JS SDK that manages the state via Redux.

There are a few things you need to set up in a React/Native project to cleverly interact with Pocketbase.

This "SDK" does it all for you.

Instead of waiting directly for a callback/response from the client, you can simply use a hook - "useAppContent" to query Redux.
Also use "actions" from "useAppContent" to subscribe, unsubscribe, re-fetch, create, update or delete.

The main functionality has been running very clean and I'm satisfied so far, but there are still a few things that need to be improved, such as auth, and error handling when connecting, (filter, sort, etc).

But all of this should be easy to integrate and shouldn't take too long to catch up.

I am happy about every pull request ✌🏻

Have fun using it 🎉

[github.com/tobicrain/pocketbase-react](https://github.com/tobicrain/pocketbase-react)
[npmjs.com/package/pocketbase-react](https://npmjs.com/package/pocketbase-react)

**Comment by benallfree** [comment]
Hey @tobicrain this looks great! I am the maintainer of https://github.com/benallfree/awesome-pocketbase, do you mind if I add your project to the directory?

**Reply by tobicrain** [reply]
Thank you 😄
yes sure, no problem 👍🏼

**Reply by kingdevnl** [reply]
It would be amazing if you could add OAuth (Social Login) support into it.

**Reply by tobicrain** [reply]
> It would be amazing if you could add OAuth (Social Login) support into it. 

Good point, I will probably integrate OAuth when improving the current Auth handling system

**Comment by tobicrain** [comment]
@kingdevnl OAuth is now implemented since version 0.1.9
I still have to adapt the readme but you have access to all auth-related stuff with the useAuth hook
